<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>505</width>
    <height>277</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>505</width>
    <height>277</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>DS_EX3_03363636_蔡璨名 Magic Square</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QWidget" name="widgetControls" native="true">
      <layout class="QVBoxLayout" name="layoutControls">
       <item>
        <layout class="QHBoxLayout" name="layoutSize">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <item>
          <widget class="QLabel" name="labelSize">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="statusTip">
            <string>Input the square size. It must be &gt;= 3</string>
           </property>
           <property name="text">
            <string>Size = </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="txtSize">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="statusTip">
            <string>Input the square size. It must be an odd number not less than 3.</string>
           </property>
           <property name="text">
            <string>3</string>
           </property>
           <property name="placeholderText">
            <string>odd num &gt;= 3</string>
           </property>
           <property name="clearButtonEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupDirection">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>203</width>
           <height>132</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>203</width>
           <height>132</height>
          </size>
         </property>
         <property name="statusTip">
          <string>Direction of the algorithm.</string>
         </property>
         <property name="title">
          <string>Direction</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QRadioButton" name="radioTopLeft">
            <property name="text">
             <string>Top-&amp;Left</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioTopRight">
            <property name="text">
             <string>&amp;Top-Right</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioBottomLeft">
            <property name="text">
             <string>&amp;Bottom-Left (Not Perfect)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioBottomRight">
            <property name="text">
             <string>Bottom-&amp;Right (Not Perfect)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item alignment="Qt::AlignHCenter|Qt::AlignTop">
        <widget class="QPushButton" name="btnExecute">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>215</width>
           <height>32</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>215</width>
           <height>32</height>
          </size>
         </property>
         <property name="statusTip">
          <string>Click to execute the magic square algorithm.</string>
         </property>
         <property name="text">
          <string>E&amp;xecute</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widgetSpacing" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTableView" name="tableView">
      <property name="statusTip">
       <string>Result of the square will be shown here.</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>505</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="toolTip">
     <string>File Menu</string>
    </property>
    <property name="statusTip">
     <string>File Menu</string>
    </property>
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="toolTip">
     <string>Help Menu</string>
    </property>
    <property name="statusTip">
     <string>Help Menu</string>
    </property>
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="toolTip">
    <string>Exit the application.</string>
   </property>
   <property name="statusTip">
    <string>Exit the application.</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="toolTip">
    <string>About this application.</string>
   </property>
   <property name="statusTip">
    <string>About this application.</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>radioTopLeft</tabstop>
  <tabstop>radioTopRight</tabstop>
  <tabstop>radioBottomLeft</tabstop>
  <tabstop>radioBottomRight</tabstop>
  <tabstop>btnExecute</tabstop>
  <tabstop>tableView</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
